redfish-core/lib/managers.hpp

```diff
@@ -1922,6 +1922,61 @@ inline void setDateTime(std::shared_ptr<bmcweb::AsyncResp> aResp,
     }
 }
 
+/**
+ * @brief Retrieves bmc state properties over dbus
+ *
+ * @param[in] aResp     Shared pointer for completing asynchronous calls.
+ *
+ * @return None.
+ */
+inline void getBMCState(const std::shared_ptr<bmcweb::AsyncResp>& aResp)
+{
+    BMCWEB_LOG_DEBUG << "Get BMC information.";
+    crow::connections::systemBus->async_method_call(
+        [aResp](const boost::system::error_code ec,
+                const std::variant<std::string>& bmcState) {
+            if (ec)
+            {
+                BMCWEB_LOG_DEBUG << "DBUS response error " << ec;
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            const std::string* s = std::get_if<std::string>(&bmcState);
+            BMCWEB_LOG_DEBUG << "BMC state: " << *s;
+            if (s != nullptr)
```

Comment:
These are typically reverse checks in bmcweb 

if (state == nullptr)
{
internalError; 
return;

}

Reply:
done